test/ecto/adapters/mysql_test.exs

```diff
@@ -161,6 +161,16 @@ defmodule Ecto.Adapters.MySQLTest do
     end
   end
 
+  test "union" do
+    other_query = Schema |> select([r], r.y)
+
+    query = Schema |> select([r], r.x) |> union(other_query) |> plan()
+    assert all(query) == ~s{SELECT s0.`x` FROM `schema` AS s0 UNION SELECT s0.`y` FROM `schema` AS s0}
+
+    query = Schema |> select([r], r.x) |> union_all(other_query) |> plan()
+    assert all(query) == ~s{SELECT s0.`x` FROM `schema` AS s0 UNION ALL SELECT s0.`y` FROM `schema` AS s0}
```

Comment:
We need to add a test for the precedence. If I do `query1 |> union_all(query2 |> union_all(query3))`, we need parens around the second pair. An easier solution is also to always add parens around the operators whenever UNION is used.

Reply:
That's true. We also can have ORDER BY, OFFSET and LIMIT in union queries and without parens it would be invalid SQL so I'm going to test this case too.